src/shm_main.c

```diff
@@ -768,6 +813,273 @@ sr_shmmain_createunlock(int shm_lock)
     }
 }
 
+/**
+ * @brief Determine whether the connection specified by cid is alive.
+ *
+ * @param[in]  cid Connection ID to check
+ * @param[out] conn_alive Set to non-zero if the connection is alive, 0 otherwise.
+ * @return err_info, NULL on success.
+ */
+sr_error_info_t *
+sr_shmmain_check_conn_lock(sr_cid_t cid, int *conn_alive) {
+    sr_error_info_t *err_info = NULL;
+    struct flock fl;
+    int fd = -1;
+    char *path = NULL;
+    int rc = 0;
+    conn_list_entry *ptr = NULL;
+
+    assert(conn_alive);
+
+    (*conn_alive) = 0;
+
+    /* Connection ID NONE is reserved and never used. */
+    if (SR_CONN_ID_NONE == cid) {
```

Comment:
Based on the comment this should output some internal error or even be an assert at the function beginning, right?

Reply:
Added an internal error.